### PR TITLE
PARALLELIZE

### PR DESCRIPTION
**Pull request by iancanderson**
Puns are generated in parallel, spawning one process per rhyme of the
original word.

Inspired by:
http://www.reactive.io/tips/2015/02/03/introduction-to-parallel-computing-with-elixir/

### CODE REVIEW

**Comment by iancanderson** [comment]
cc @gabebw

File: lib/elixirls_just_want_to_have_puns.ex [hunk 1]
```diff
@@ -28,15 +28,51 @@ defmodule ElixirlsJustWantToHavePuns do
...
+    puns
+    |> Enum.each &(IO.puts(&1))
+  end
+end
+
+defmodule PunMaker do
+  import Enum
+
+  def spawn_make(pid, original_word, rhymebrain_result) do
+    spawn __MODULE__, :send_make, [pid, original_word, rhymebrain_result]
```
Comment (gabebw) [inline lib/elixirls_just_want_to_have_puns.ex:63]: Whoa, what's happening here? I don't actually know elixir, I just know puns

File: lib/elixirls_just_want_to_have_puns.ex [hunk 2]
```diff
@@ -28,15 +28,51 @@ defmodule ElixirlsJustWantToHavePuns do
...
+
+    receive_puns(length(rhymebrain_results), [])
+  end
+
+  defp receive_puns(length, result) do
+    receive do
+      puns ->
+        result = puns ++ result
+
+        if length == 1 do
```
Comment (drapergeek) [inline lib/elixirls_just_want_to_have_puns.ex]: Could you define `receive_puns`, with two different implementations and have where that does this when `length==1`? That way this always just called `receive_puns`?

File: lib/elixirls_just_want_to_have_puns.ex (hunk 1 above)
Comment (gabebw) [inline lib/elixirls_just_want_to_have_puns.ex:63]: Oh I read the blog post. I get it now.

File: lib/elixirls_just_want_to_have_puns.ex (hunk 2 above)
Comment (iancanderson) [inline lib/elixirls_just_want_to_have_puns.ex]: ooohhh I like that. will work on that

**Comment by BlakeWilliams** [comment]
Does this mean that the repo should be renamed to `Girls just want to parallelize puns`?

**Comment by iancanderson** [comment]
thanks everyone

**Pull request closed without merging**